test/SlowTests/Server/NotificationCenter/OutOfMemoryNotificationTests.cs

```diff
@@ -0,0 +1,165 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using FastTests;
+using Raven.Server.NotificationCenter;
+using Raven.Server.NotificationCenter.Notifications;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+using Sparrow.Server.Collections;
+using Xunit;
+
+namespace SlowTests.Server.NotificationCenter
+{
+    public class OutOfMemoryNotificationTests : RavenLowLevelTestBase
+    {
+        [Fact]
+        public void Add_WhileAllHaveTheSameKey_ShouldRemindOnlyOne()
+        {
+            using (var database = CreateDocumentDatabase())
+            {
+                var environment = database.GetAllStoragesEnvironment().First().Environment;
+
+                var actions = new AsyncQueue<DynamicJsonValue>();
+
+                var writer = new TestWebSocketWriter();
+                var taskList = new List<Task>();
+                using (database.NotificationCenter.TrackActions(actions, writer))
+                {
+
+                    taskList.Add(Task.Run( () => database.NotificationCenter.OutOfMemory.Add(environment, new OutOfMemoryException())));
```

Comment:
I think that we should also test that the environment is removed from the OutOfMemory collection by forcing GC

Reply:
@haludi ?